File: 12. Integer to Roman/12. Integer to Roman.rs
impl Solution {
    pub fn int_to_roman(num: i32) -> String {
        let mut result = String::new();
        let values = vec![
            (1000, "M"),
            (900, "CM"),
            (500, "D"),
            (400, "CD"),
            (100, "C"),
            (90, "XC"),
            (50, "L"),
            (40, "XL"),
            (10, "X"),
            (9, "IX"),
            (5, "V"),
            (4, "IV"),
            (1, "I"),
        ];
        
        let mut remainder = num;
        
        for (value, symbol) in values {
            while remainder >= value {
                result.push_str(symbol);
                remainder -= value;
            }
        }
        
        result
    }
}

File: 12. Integer to Roman/12. Integer to Roman.go
func intToRoman(num int) string {
    values := []int{1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1}
    symbols := []string{"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"}
    var result strings.Builder
    
    for i := 0; i < len(values) && num > 0; i++ {
        for num >= values[i] {
            result.WriteString(symbols[i])
            num -= values[i]
        }
    }
    
    return result.String()
}